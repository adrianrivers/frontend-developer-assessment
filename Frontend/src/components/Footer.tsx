import React from 'react'
import { Box, Center, Text, Link } from '@chakra-ui/react'

export default function Footer(): JSX.Element {
  return (
    <Box as="footer" width="full">
      <Center paddingY={16}>
        <Text>
          Â© 2022 Copyright <Link href="https://clearpoint.digital">clearpoint.digital</Link>
        </Text>
      </Center>
    </Box>
  )
}
