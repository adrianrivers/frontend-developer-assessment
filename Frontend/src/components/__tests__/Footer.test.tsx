import { render } from '@testing-library/react'
import React from 'react'
import Footer from '../Footer'

describe('Footer', () => {
  it('should render a Footer component', () => {
    const { container, getByText } = render(<Footer />)
    const link = getByText('clearpoint.digital').closest('a') as HTMLAnchorElement

    expect(link.href).toBe('https://clearpoint.digital/')
    expect(link.innerHTML).toBe('clearpoint.digital')
    expect(container).toMatchSnapshot()
  })
})
