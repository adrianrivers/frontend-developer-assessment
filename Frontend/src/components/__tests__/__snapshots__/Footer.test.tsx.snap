// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render a Footer component 1`] = `
<div>
  <footer
    class="css-13rbvqk"
  >
    <div
      class="css-caen9j"
    >
      <p
        class="chakra-text css-0"
      >
        Â© 2022 Copyright 
        <a
          class="chakra-link css-0"
          href="https://clearpoint.digital"
        >
          clearpoint.digital
        </a>
      </p>
    </div>
  </footer>
</div>
`;
